src/legacy/ui/public/field_editor/field_editor.js

```diff
@@ -331,6 +335,57 @@ export class FieldEditorComponent extends PureComponent {
     );
   }
 
+  /**
+   * renders a warning and a table of conflicting indices
+   * in case there are indices with different types
+   */
+  renderTypeConflict() {
+    const { intl } = this.props;
+    const { field = {} } = this.state;
+    if (!field.conflictDescriptions || !typeof field.conflictDescriptions === 'object') {
```

Comment:
tiny nitpick - `!typeof field.conflictDescriptions === 'object'` is the only conditional needed as `typeof field.conflictDescriptions` will return `undefined`

Reply:
wanted to rule out that conflictDescriptions is `null`, since that would also be an 'object', however, diving a little bit deeper:

https://github.com/elastic/kibana/blob/03624a1c12e09b5cff59f859161b6b07466a2f5f/src/legacy/server/index_patterns/service/lib/field_capabilities/field_caps_response.js#L109

`null` can be ruled out, so yes, i can be simplified

Reply:
Super minor nitpick, but can we also write this as `typeof field.conflictDescriptions !== 'object'`? This reads to me as "check that the type of `field.conflictDescriptions` IS NOT 'object'".

Reply:
@kertal thanks for reminding me that `typeof null === 'object'`

Reply:
@mattkime one of that nice JS traps/oddities like `typeof NaN === 'number'` :)
anyway, we wouldn't discuss about that, if the code was already in typescript, however, then there would be much more to review, so better in a separate step/PR

Reply:
done